src/coreclr/nativeaot/Runtime/eventtrace.cpp

```diff
@@ -94,6 +94,42 @@ enum CallbackProviderIndex
     DotNETRuntimePrivate = 3
 };
 
+#ifdef FEATURE_ETW
+// EventFilterType identifies the filter type used by the PEVENT_FILTER_DESCRIPTOR
+enum EventFilterType
+{
+    None = 0x0,
+    ClientSequenceNumber = 0x1,
+    // EventPipePayload = 0x2,
+    // Schematized = 0x80000000,
+    // SystemFlags = 0x80000001,
+    // TraceHandle = 0x80000002,
+    // PID = 0x80000004,
+    // ExecutableName = 0x80000008,
+    // PackageID = 0x80000010,
+    // PackageAppID = 0x80000020,
+    // Payload = 0x80000100,
+    // EventID = 0x80000200,
+    // EventName = 0x80000400,
+    // Stackwalk = 0x80001000,
+    // StackwalkName = 0x80002000,
+    // StackwalkLevelKW = 0x80004000,
+};
+
+void ParseFilterDataClientSequenceNumber(
+    EVENT_FILTER_DESCRIPTOR * FilterData,
+    LONGLONG * pClientSequenceNumber)
+{
+    if (FilterData == NULL)
+        return;
+
+    if (FilterData->Type == ClientSequenceNumber && FilterData->Size == sizeof(LONGLONG))
+    {
+        *pClientSequenceNumber = *(LONGLONG *) (FilterData->Ptr);
+    }
```

Comment:
NativeAOT supports EventPipe too so unless there is particular reason not to I'd support the same parsing options that we have on CoreCLR.

Reply:
I left it out since there was a comment that `// NativeAOT currently only supports forcing a GC with ManagedHeapCollectKeyword via ETW`, so this path won't get hit until forcing a GC via EventPipe is supported on NativeAOT. I figured the parsing could be added once/if the EventPipe scenario is supported.

I don't mind adding the parsing, if that is preferred. @LakshanF @elinor-fung do y'all happen to know if NativeAOT will ever support forcing a GC with the keyword via EventPipe?

Reply:
I'm not sure why that limitation is there and I'd be pretty surprised if its fundamental. My guess was that it was omitted just for lack of time to implement/test something.

Reply:
Agreed with @noahfalk here.  I suspect this was an exercise in scoping more than anything else.

Reply:
Thanks! Added the EventPipe parsing to the nativeaot counterpart.